.buildkite/commands/build-ios.sh

```diff
@@ -7,13 +7,10 @@ echo '--- :ios: Set env var for iOS E2E testing'
 set -x
 export TEST_RN_PLATFORM=ios
 export TEST_ENV=sauce
-CONFIG_FILE="$(pwd)/gutenberg/packages/react-native-editor/__device-tests__/helpers/device-config.json"
-# Uses the local deviceName since SauceLabs uses a different one.
-IOS_DEVICE_NAME=$(jq -r '.ios.local.deviceName' "$CONFIG_FILE")
-IOS_PLATFORM_VERSION=$(jq -r '.ios.buildkite.platformVersion' "$CONFIG_FILE")
-# Set a destination different from the hardcoded one which only works in the
-# older Xcode-setup used by CircleCI
-export RN_EDITOR_E2E_IOS_DESTINATION="platform=iOS Simulator,name=$IOS_DEVICE_NAME,OS=$IOS_PLATFORM_VERSION"
+# We must use a simulator that's available on the selected Xcode version
+# otherwsie Xcode fallbacks to "generic destination" which requires provision
+# profiles to built the Demo app.
+export RN_EDITOR_E2E_IOS_DESTINATION="platform=iOS Simulator,name=iPhone 15,OS=17.2"
```

Comment:
@mokagio I didn't change [the `buildkite.platformVersion`](https://github.com/WordPress/gutenberg/blob/e8b0a4c459dd2dd8f6775f312128cc6e7106a6bc/packages/react-native-editor/__device-tests__/helpers/device-config.json#L13), because I think its value depends on the Xcode version used in the pipeline or on our own mac. I'm not sure if we need the dance of updating it in the `wordpress/gutenberg` repo so that pipeline in this repo can pass. I'm not sure if I missed anything though...

Reply:
Thanks @crazytonyli. Yeah I'm not sure either... I think this hardcoded solution is an excellent step to get this over the line.

We can then think about the best type of configuration.

However, I too feel uncomfortable with having to set values for a CI build that runs on this repo via a different repo. Given this handy env var that allows overrides, I think we could use this solution in the long run, too.